stac_fastapi/pgstac/app.py

```diff
@@ -123,17 +155,31 @@
         extensions=itm_col_extensions,
         request_type="GET",
     )
+    application_extensions.extend(itm_col_extensions)
 
 # /collections model
 collections_get_request_model = EmptyRequest
 if "collection_search" in enabled_extensions:
-    cs_extensions = [
-        extension
-        for key, extension in cs_extensions_map.items()
-        if key in enabled_extensions
+    mixins = []
+    mixing_conformances = [
+        "https://api.stacspec.org/v1.0.0-rc.1/collection-search",
+        "http://www.opengis.net/spec/ogcapi-common-2/1.0/conf/simple-query",
     ]
-    collection_search_extension = CollectionSearchExtension.from_extensions(cs_extensions)
-    collections_get_request_model = collection_search_extension.GET
+    for key, extension in cs_extensions_map.items():
+        if key not in enabled_extensions:
+            continue
+        mixins.append(extension.GET)
+        mixing_conformances.extend(extension.conformance_classes)
+
+    collections_get_request_model = create_request_model(
+        model_name="CollectionsGetRequest",
+        base_model=BaseCollectionSearchGetRequest,
+        mixins=mixins,
+        request_type="GET",
+    )
+    collection_search_extension = CollectionSearchExtension(
+        GET=collections_get_request_model, conformance_classes=mixing_conformances
+    )
```

Comment:
we cannot use `CollectionSearchExtension.from_extensions(cs_extensions)` because `from_extensions` will look for `FilterExtension` class name in the ext but we have `CollectionSearchFilterExtension` now 😅 

https://github.com/stac-utils/stac-fastapi/blob/5ded0c685d50ae6b3c1af6ff039529e631dc2716/stac_fastapi/extensions/stac_fastapi/extensions/core/collection_search/collection_search.py#L93-L95

Reply:
maybe we could make the `known_extension_conformances` an attribute of `CollectionSearchExtension` that could be specified by the user so we don't need to re-write the `from_extensions` method here.